livesync_citadel/indico_livesync_citadel/search.py

```diff
@@ -0,0 +1,40 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2020 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+import requests
+from werkzeug.urls import url_join
+
+from indico_livesync_citadel.plugin import LiveSyncCitadelPlugin
+from indico_livesync_citadel.models.search_id_map import EntryType
+
+
+# https://cern-search.docs.cern.ch/usage/operations/#query-documents
+# this will fetch 
+def search(query, object_type=EntryType.event, page=1):
+    token = LiveSyncCitadelPlugin.settings.get('search_backend_token')
+    search_backend_url = LiveSyncCitadelPlugin.settings.get('search_backend_url')
+    query_url = url_join(search_backend_url, 'api/records/')
+
+    # this token is used by the backend to authenticate and also to filter
+    # the objects that we can actually read
+    headers = {
+        'Authorization': 'Bearer {}'.format(token)
+    }
+
+    # look for objects matching the `query` and schema
+    # make sure the query is properly escaped (https://cern-search.docs.cern.ch/usage/operations/#advanced-queries)
+    q = '+{} +$schema:*{}*'.format(query, object_type.name)
+    params = {'page': page, 'q': q}
+    resp = requests.get(query_url, params=params, headers=headers)
+    if not resp.ok:
+        return []
+
+    data = resp.json()['hits']
+    total, objects = data['total'], data['hits']
```

Comment:
total is the total number from all pages?

Reply:
Total is the number of all objects that match the criteria.

Reply:
ok that's what i thought :)